api/v1/controllers/collectors.js

```diff
@@ -285,6 +283,24 @@ function heartbeat(req, res, next) {
     return o.save();
   })
 
+  // get the ids for changed generators
+  .then(() => heartbeatUtils.getChanges(collectorName))
+
+  // find the generator objects for the changes ids
+  .then((changed) => Promise.join(
+      Generator.findAll({ where: { id: changed.added } }),
```

Comment:
why three separate db queries here?

Reply:
I need to be able to distinguish between the added, deleted, and updated generators sent back from the db.

Reply:
So your assumption is that three simultaneous db queries is "cheaper" than using one query and then iterating over the results to assign each result to the right set based on the id?

Reply:
No, just that it's cleaner this way. I can do that if you're concerned about the performance.

Reply:
I'm okay with doing it either way, but please document your reasoning in a code comment.

Reply:
Note that in practice it will be rare for multiple generators to be updated in different ways for the same collector within the same heartbeat cycle so most of the time there will only be one query sent anyway.